src/db.rs

```diff
@@ -31,6 +31,7 @@ pub(crate) struct DbInner {
     pub(crate) table_store: Arc<TableStore>,
     pub(crate) memtable_flush_notifier: tokio::sync::mpsc::UnboundedSender<MemtableFlushThreadMsg>,
     pub(crate) db_stats: Arc<DbStats>,
+    pub(crate) sst_format: SsTableFormat,
```

Comment:
Was debating `Arc<SsTableFormat>` instead. I think I prefer the way you have it since SsTableFormat is just a usize, u32, and enum. Cloning is OK.

I also think keeping the format in `DbInner` makes sense.